### ISSUE TITLE
Load roslyn version of the codestyle analyzers to fix torn load state (causes ReflectionTypeLoadException)

### ISSUE THREAD

**Issue by amcasey** [dotnet/roslyn#72811]
**Version Used**: 4.10.0-3.24175.2+919d4dbfb0dffb35a702417e28ceea652d248bc6

**Steps to Reproduce**: Trigger CSharpUseCollectionInitializerCodeFixProvider

**Expected Behavior**:

Fix applied

**Actual Behavior**:

`"Method 'get_IsParamsArray' in type 'Microsoft.CodeAnalysis.CodeGeneration.CodeGenerationParameterSymbol' from assembly 'Microsoft.CodeAnalysis.CodeStyle.Fixes, Version=4.10.10.15805, Culture=neutral, PublicKeyToken=31bf3856ad364e35' does not have an implementation."	string`

**Comment by sharwell** [comment]
This is occurring because CodeGenerationParameterSymbol in the code style layer breaks Roslyn's rule that the code style layer must only reference public APIs. `ISymbol` does not allow public implementations:

https://github.com/dotnet/roslyn/blob/0f35ec3b83a0291003bb64fd7b78de0d423d704f/src/Compilers/Core/Portable/Symbols/ISymbol.cs#L22-L23

This type must be removed from the code style layer and/or updated to no longer implement `ISymbol`.

**Comment by CyrusNajmabadi** [comment]
> ISymbol does not allow public implementations:

That's not the case. Isymbol does not allow 3rd party impls. It allows first party impls.  I know, I created it for that purpose :-)

When this interface was updated, we appropriately updated all our apis.  We need to ship that updated API now.

**Comment by abbottdev** [comment]
Same issue here, using latest preview of the C# DevKit (`v1.5.14 (pre-release)`) and C# `v2.26.13 (pre-release)`. I'm using Apple Silicon, .NET 8.

```
Method 'get_IsParamsArray' in type 'Microsoft.CodeAnalysis.CodeGeneration.CodeGenerationParameterSymbol' from assembly 'Microsoft.CodeAnalysis.CodeStyle.Fixes, Version=4.9.10.12906, Culture=neutral, PublicKeyToken=31bf3856ad364e35' does not have an implementation.
   at System.Reflection.RuntimeModule.GetTypes(RuntimeModule module)
   at System.Composition.Hosting.ContainerConfiguration.<>c.<WithAssemblies>b__16_0(Assembly a)
   at System.Linq.Enumerable.SelectManySingleSelectorIterator`2.MoveNext()
   at System.Composition.TypedParts.TypedPartExportDescriptorProvider..ctor(IEnumerable`1 types, AttributedModelProvider attributeContext)
   at System.Composition.Hosting.ContainerConfiguration.CreateContainer()
   at Microsoft.CodeAnalysis.Host.CodeStyleHostLanguageServices.MefHostExportProvider.Create(String languageName)
   at System.Runtime.CompilerServices.ConditionalWeakTable`2.GetValueLocked(TKey key, CreateValueCallback createValueCallback)
   at Microsoft.CodeAnalysis.Host.CodeStyleHostLanguageServices..ctor(HostLanguageServices hostLanguageServices)
   at Microsoft.CodeAnalysis.Host.CodeStyleHostLanguageServices.Create(HostLanguageServices hostLanguageServices)
   at System.Runtime.CompilerServices.ConditionalWeakTable`2.GetValueLocked(TKey key, CreateValueCallback createValueCallback)
   at Microsoft.CodeAnalysis.Host.CodeStyleHostLanguageServices.GetRequiredMappedCodeStyleLanguageServices(HostLanguageServices hostLanguageServices)
   at Microsoft.CodeAnalysis.RemoveUnnecessaryImports.AbstractRemoveUnnecessaryImportsCodeFixProvider.RemoveUnnecessaryImportsAsync(Document document, CodeActionOptionsProvider fallbackOptions, CancellationToken cancellationToken)
   at Microsoft.CodeAnalysis.CodeActions.CodeAction.GetChangedSolutionAsync(IProgress`1 progress, CancellationToken cancellationToken) in /_/src/Workspaces/Core/Portable/CodeActions/CodeAction.cs:line 356
   at Microsoft.CodeAnalysis.CodeActions.CodeAction.ComputeOperationsAsync(IProgress`1 progress, CancellationToken cancellationToken) in /_/src/Workspaces/Core/Portable/CodeActions/CodeAction.cs:line 310
   at Microsoft.CodeAnalysis.CodeActions.CodeAction.GetOperationsCoreAsync(Solution originalSolution, IProgress`1 progress, CancellationToken cancellationToken) in /_/src/Workspaces/Core/Portable/CodeActions/CodeAction.cs:line 247
   at Microsoft.CodeAnalysis.LanguageServer.Handler.CodeActionResolveHandler.HandleRequestAsync(CodeAction codeAction, RequestContext context, CancellationToken cancellationToken) in /_/src/Features/LanguageServer/Protocol/Handler/CodeActions/CodeActionResolveHandler.cs:line 92
   at Microsoft.CommonLanguageServerProtocol.Framework.QueueItem`3.StartRequestAsync(TRequestContext context, IMethodHandler handler, CancellationToken cancellationToken) in /_/src/Features/LanguageServer/Microsoft.CommonLanguageServerProtocol.Framework/QueueItem.cs:line 150
System.TypeLoadException: Method 'get_IsParamsArray' in type 'Microsoft.CodeAnalysis.CodeGeneration.CodeGenerationParameterSymbol' from assembly 'Microsoft.CodeAnalysis.CodeStyle.Fixes, Version=4.9.10.12906, Culture=neutral, PublicKeyToken=31bf3856ad364e35' does not have an implementation.
[Error - 15:16:11] Request codeAction/resolve failed.
  Message: Unable to load one or more of the requested types.
Method 'get_IsParamsArray' in type 'Microsoft.CodeAnalysis.CodeGeneration.CodeGenerationParameterSymbol' from assembly 'Microsoft.CodeAnalysis.CodeStyle.Fixes, Version=4.9.10.12906, Culture=neutral, PublicKeyToken=31bf3856ad364e35' does not have an implementation.
  Code: -32000 
[object Object]
```

**Comment by AnthonyMastrean** [comment]
Affects the misplaced using directives quick fix / code action, too.

```
[Error - 3:13:33 PM] [LanguageServerExtensionManager] Microsoft.CodeAnalysis.CSharp.MisplacedUsingDirectives.MisplacedUsingDirectivesCodeFixProvider threw an exception. System.Reflection.ReflectionTypeLoadException: Unable to load one or more of the requested types.
Method 'get_IsParamsArray' in type 'Microsoft.CodeAnalysis.CodeGeneration.CodeGenerationParameterSymbol' from assembly 'Microsoft.CodeAnalysis.CodeStyle.Fixes, Version=4.7.9.1717, Culture=neutral, PublicKeyToken=31bf3856ad364e35' does not have an implementation.
   at System.Reflection.RuntimeModule.GetTypes(RuntimeModule module)
   at System.Composition.Hosting.ContainerConfiguration.<>c.<WithAssemblies>b__16_0(Assembly a)
   at System.Linq.Enumerable.SelectManySingleSelectorIterator`2.MoveNext()
   at System.Composition.TypedParts.TypedPartExportDescriptorProvider..ctor(IEnumerable`1 types, AttributedModelProvider attributeContext)
   at System.Composition.Hosting.ContainerConfiguration.CreateContainer()
   at Microsoft.CodeAnalysis.Host.CodeStyleHostLanguageServices.MefHostExportProvider.Create(String languageName)
   at System.Runtime.CompilerServices.ConditionalWeakTable`2.GetValueLocked(TKey key, CreateValueCallback createValueCallback)
   at Microsoft.CodeAnalysis.Host.CodeStyleHostLanguageServices..ctor(HostLanguageServices hostLanguageServices)
   at Microsoft.CodeAnalysis.Host.CodeStyleHostLanguageServices.Create(HostLanguageServices hostLanguageServices)
   at System.Runtime.CompilerServices.ConditionalWeakTable`2.GetValueLocked(TKey key, CreateValueCallback createValueCallback)
   at Microsoft.CodeAnalysis.Host.CodeStyleHostLanguageServices.GetRequiredMappedCodeStyleLanguageServices(HostLanguageServices hostLanguageServices)
   at Microsoft.CodeAnalysis.CodeActions.CSharpCodeFixOptionsProviders.GetCSharpCodeFixOptionsProviderAsync(Document document, CodeActionOptionsProvider fallbackOptions, CancellationToken cancellationToken)
   at Microsoft.CodeAnalysis.CSharp.MisplacedUsingDirectives.MisplacedUsingDirectivesCodeFixProvider.RegisterCodeFixesAsync(CodeFixContext context)
   at Microsoft.CodeAnalysis.CodeFixes.CodeFixService.GetCodeFixesAsync(TextDocument document, TextSpan span, CodeFixProvider fixer, CodeChangeProviderMetadata fixerMetadata, CodeActionOptionsProvider fallbackOptions, ImmutableArray`1 diagnostics, Dictionary`2 uniqueDiagosticToEquivalenceKeysMap, Dictionary`2 diagnosticAndEquivalenceKeyToFixersMap, CancellationToken cancellationToken) in /_/src/Features/LanguageServer/Protocol/Features/CodeFixes/CodeFixService.cs:line 661
   at Microsoft.CodeAnalysis.Extensions.IExtensionManagerExtensions.PerformFunctionAsync[T](IExtensionManager extensionManager, Object extension, Func`2 function, T defaultValue, CancellationToken cancellationToken) in /_/src/Workspaces/Core/Portable/ExtensionManager/IExtensionManagerExtensions.cs:line 84
System.TypeLoadException: Method 'get_IsParamsArray' in type 'Microsoft.CodeAnalysis.CodeGeneration.CodeGenerationParameterSymbol' from assembly 'Microsoft.CodeAnalysis.CodeStyle.Fixes, Version=4.7.9.1717, Culture=neutral, PublicKeyToken=31bf3856ad364e35' does not have an implementation.
```

**Comment by kylepope-ge** [comment]
I have been having this exact problem on my M1 Mac as well. Happens on pretty much any codeAction which makes those unable to be used entirely. I'm using Visual Studio Code 1.89.0, C# extension 2.28.11, C# Dev Kit extension 1.5.20. Error messages look like the following every time:
```
[Error - 9:38:55 AM] Request codeAction/resolve failed.
  Message: Unable to load one or more of the requested types.
Method 'get_IsParamsArray' in type 'Microsoft.CodeAnalysis.CodeGeneration.CodeGenerationParameterSymbol' from assembly 'Microsoft.CodeAnalysis.CodeStyle.Fixes, Version=4.8.9.2411, Culture=neutral, PublicKeyToken=31bf3856ad364e35' does not have an implementation.
  Code: -32000 
[object Object]
```

**Comment by kylepope-ge** [comment]
More investigation this morning on this `get_IsParamsArray` error... I discovered this problem is with the C# extension (I disabled my other C# Dev Kit and related extensions). The problem started happening in the `2.23.15` release. The previous release (`2.22.5`) doesn't have the problem, so I'm going to downgrade for now.

**Comment by Tum4ik** [comment]
I have the same problem when trying to get SyntaxGenerator for my source generation purposes:
```csharp
try
{
  var syntaxGenerator = SyntaxGenerator.GetGenerator(new AdhocWorkspace(), LanguageNames.CSharp);
}
catch (ReflectionTypeLoadException e)
{
  throw;
}
```

**Comment by sharwell** [comment]
@Tum4ik Yours is a separate issue even if it's the same exception. It's caused by attempting to load a different version of Microsoft.CodeAnalysis.dll from the version of Microsoft.CodeAnalysis.Workspaces.dll, which is not supported.

**Comment by mcallaghan-geotab** [comment]
what is the workaround?

**Comment by sharwell** [comment]
There is no workaround currently. These features are currently broken even for ourselves.

**Comment by Tum4ik** [comment]
@mcallaghan-geotab I have Visual Studio 17.9 on my another PC, I didn't update that to 17.10 and I don't have the problem there. But @sharwell mentioned I have a different problem. Who knows... maybe you can downgrade to 17.9.

**Comment by mcallaghan-geotab** [comment]
(experiencing the problem in Rider fwiw - not visual studio)

**Comment by Tum4ik** [comment]
Updating `Microsoft.CodeAnalysis.CSharp.Workspaces` package from `4.9.2` to `4.10.0` in my source generator project has fixed the problem.

**Comment by smkanadl** [comment]
@sharwell Do I understand correctly that this won't be fixed asap and I have to wait for the 17.11 release instead? If so, it is really disappointing!

**Comment by ShreyasJejurkar** [comment]
Why is this still not fixed? It's blocking a good experience for Visual Studio Code users.

**Comment by CyrusNajmabadi** [comment]
@ShreyasJejurkar because the solution is complex and is taking time.

**Comment by smkanadl** [comment]
> Why is this still not fixed? It's blocking a good experience for Visual Studio Code users.

And for all Visual Studio Users. Which is annoying as it is not cheap at all.

**Comment by smkanadl** [comment]
> @ShreyasJejurkar because the solution is complex and is taking time.

If breaking it was that easy and underwent unnoticed, fixing it (especially for a payed product) cannot take 4+ month. That's simply not acceptable.

**Comment by airbreather** [comment]
We worked around it by turning off `EnforceCodeStyleInBuild` for now. `dotnet format --verify-no-changes` is almost as good for us, and this issue did seem to completely disappear once we did, so I've been content with waiting for a fix to be done properly.

**Comment by CyrusNajmabadi** [comment]
> fixing it (especially for a payed product) cannot take 4+ month

The fix is not simple.  And it will take time to get working. 

> fixing it (especially for a payed product) cannot take 4+ month

there is no relation on the two.  fixes take the time needed to fix them.  In this case due to the size and complexity of the fix, that means it takes a while.

--

Fundamentally, the issue is due to your setup loading DLLs that are not compatible with binary changes made in VS.  We have no workaround on that until new versions of those DLLs ship that actually work with VS, or we mandate that we will not load dlls from your setup.  These are non trivial pieces of work.

**Comment by smkanadl** [comment]
> > fixing it (especially for a payed product) cannot take 4+ month
> 
> The fix is not simple. And it will take time to get working.
> 
> > fixing it (especially for a payed product) cannot take 4+ month
> 
> there is no relation on the two. fixes take the time needed to fix them. In this case due to the size and complexity of the fix, that means it takes a while.
> 
> --
> 
> Fundamentally, the issue is due to your setup loading DLLs that are not compatible with binary changes made in VS. We have no workaround on that until new versions of those DLLs ship that actually work with VS, or we mandate that we will not load dlls from your setup. These are non trivial pieces of work.

**I already understood that it is an complex issue**. But again: It was done with one release and no quality check was triggered during the release process that basically all "productivity features" are broken.

This is very surprising. And obviously it is not "my setup" loading the wrong dlls, but many people's setups loading the front dlls.

I understand that there may be no quick fix. But for me (and probably many others) a quite expensive tool lost many of its features.

And it's not that Microsoft is just a 2 man company that struggles with manpower to maintain VS. And my opinion is still: Breaking a payed product and not fixing it for 4+ month. Especially if going back is not easy and would also require to go back in SDK version etc.

**Comment by CyrusNajmabadi** [comment]
> It was done with one release and no quality check was triggered during the release process that basically all "productivity features" are broken.

All quality checks passed because we test VS.  We do not test every single potential configuration out there. In this case, your configuration pulls in analyzers that were built against an older Roslyn version, despite the version of Roslyn in vs moving forward. 

Roslyn, as it is expected to do, made intentional binary changes. In this case, adding new required apis on some interfaces related to new language features. We do this all the time. It is a requirement that downstream apis implement those interfaces if they want to get loaded. 

> I understand that there may be no quick fix. But for me (and probably many others) a quite expensive tool lost many of its features.

Yes.  And we are working on a solution. But, as mentioned, it is not simple or easy. 

> And it's not that Microsoft is just a 2 man company that struggles with manpower to maintain VS. 

The solution is not easy. .net itself requires that if an interface has members that those members be implemented in the types implementing those members. 

Fundamentally, this is a request to somehow make that work. Something vs has never supported. We are working on a way to get your particular setup working. But it is non trivial and requires a lot of discussion, implementation time, and validation.

**Comment by smkanadl** [comment]
> > It was done with one release and no quality check was triggered during the release process that basically all "productivity features" are broken.
> 
> All quality checks passed because we test VS. We do not test every single potential configuration out there. In this case, your configuration pulls in analyzers that were built against an older Roslyn version, despite the version of Roslyn in vs moving forward.

I would really like to understand what kind of configuration triggers this issue. And if it is a configuration, I must be very unlucky to be affected on multiple devices and it seems unlikely that a configuration issue has no workaround.

So what configuration do I need to properly use my VS?

**Comment by smkanadl** [comment]
> > So what configuration do I need to properly use my VS?
> 
> I think it's the use of `EnforceCodeStyleInBuild`. It tries to load the older version of some analyzers from packages you have loaded _into_ vs, where they don't abide by these changed interfaces.
> 
> If you disable that, we won't try to load them and things should hopefully be ok

I gave it a quick try, removed it from my `Directory.Build.props` file and indeed the tested code fixes work again.

Why is this not listed as a workaround somewhere? It's fair from ideal, but I can still conditionally enable it for CI builds to enforce clean, no warnings builds.

Many thanks for pointing this out 👍

**Comment by airbreather** [comment]
> Why is this not listed as a workaround somewhere? It's fair from ideal, but I can still conditionally enable it for CI builds to enforce clean, no warnings builds.

I posted https://github.com/dotnet/roslyn/issues/72811#issuecomment-2248715193 just as I saw the latest round of comments and noticed that nobody had explicitly posted that on this thread.

**Comment by smkanadl** [comment]
17.11 is out and the issue is still not fixed. What is the ETA?

**Comment by patoJamriska** [comment]
Having the same issue. tried modifying the sstyle packages and it is a no go. Momentarily using copilot as well to do code checks if needed, but can't use the quick tools for code cleanup on the UI of vscode.

**Comment by ericstj** [comment]
The bug here is that `IParameterSymbol`, a publicly shipping interface, added a new property (without a default implementation):
https://github.com/dotnet/roslyn/commit/1c5c7e2bfa779ba129847cfb694d22dff21cc4f5#diff-fb4bb69624c34616c08e1dc2f8236bc80b67239dd33624cd764e38976413ace9R38

That's a binary breaking change.  It will break anyone that implemented `IParameterSymbol` before that change.

To workaround this you need to update the assembly that contains the `IParameterSymbol` - in this case `Microsoft.CodeAnalysis.CodeStyle.Fixes` to a version that has reacted to the binary breaking change in `IParameterSymbol`.

**Comment by ericstj** [comment]
The attached repro demonstrates this binary-breaking change:
[codeAnalysisRepro.zip](https://github.com/user-attachments/files/16935394/codeAnalysisRepro.zip)

```xml
<ItemGroup>
    <!-- compile against older CodeAnalysis --> 
    <PackageDownload Include="Microsoft.CodeAnalysis.Common" Version="[4.8.0]" />
    <Reference Include="$(NuGetPackageRoot)\microsoft.codeanalysis.common\4.8.0\lib\netstandard2.0\Microsoft.CodeAnalysis.dll" />
    <!-- run against newer one -->
    <PackageReference Include="Microsoft.CodeAnalysis" Version="4.11.0" ExcludeAssets="Compile"/>
</ItemGroup>
```


```c#
var mySymbol = new MyParameterSymbol();
Console.WriteLine(((IParameterSymbol)mySymbol).Name);

#pragma warning disable RS1009  // suppress the warning to repro components allowed to implement IParameterSymbol
public class MyParameterSymbol : IParameterSymbol
{
  // IParameterSymbol implementation
}
#pragma  warning restore RS1009
```

run it:
```
Unhandled exception. System.TypeLoadException: Method 'get_IsParamsArray' in type 'MyParameterSymbol' from assembly 'codeAnalysisRepro, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null' does not have an implementation.
   at Program.<Main>$(String[] args)
```

**Comment by CyrusNajmabadi** [comment]
The only supported scenario (currently) is using a version of Roslyn sdk built against the version of Roslyn your ide uses. 

We are working on making it so this requirement is no longer in place.  But it will change such that it is irrelevant which version you reference. You will get the version in your ide.

**Comment by sharwell** [comment]
@CyrusNajmabadi I believe you mean "a version of .NET SDK where the built-in Roslyn version matches the version of Roslyn your IDE uses". When using the Roslyn SDK, e.g. for creating an analyzer, this limitation does not apply to the chosen Roslyn SDK version.

**Comment by JoeRobich** [comment]
@sharwell Can we have the design time build disable `EnforceCodeStyleOnBuild`? This should stop the SDK from adding the M.CA.CodeStyle analyzers. VS and VS Code would both then use the analyzers from the .Features layer, right?

**Comment by JeremyCaron** [comment]
This thread has been spiraling since March 29th.  Are there concrete plans for a fix here soon?

**Comment by CyrusNajmabadi** [comment]
The fix is in progress here https://github.com/dotnet/roslyn/pull/75250

**Comment by canselcik** [comment]
Added `EnforceCodeStyleInBuild=false`, my builds have been working anyway but I can't generate migrations. Pinning the SDK to 8.0.205 didn't help either.

Basically everything worked fine until I had to touch this thing called Entity Framework. I don't even want to use Entity Framework.

```
$ dotnet ef migrations add TrackedOrderIds
System.Reflection.ReflectionTypeLoadException: Unable to load one or more of the requested types.
Method 'get_IsParamsArray' in type 'Microsoft.CodeAnalysis.CodeGeneration.CodeGenerationParameterSymbol' from assembly 'Microsoft.CodeAnalysis.Workspaces, Version=4.5.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35' does not have an implementation.
   at System.Reflection.RuntimeModule.GetTypes(RuntimeModule module)
   at System.Composition.Hosting.ContainerConfiguration.<>c.<WithAssemblies>b__16_0(Assembly a)
   at System.Linq.Enumerable.SelectManySingleSelectorIterator`2.MoveNext()
   at System.Composition.TypedParts.TypedPartExportDescriptorProvider..ctor(IEnumerable`1 types, AttributedModelProvider attributeContext)
   at System.Composition.Hosting.ContainerConfiguration.CreateContainer()
   at Microsoft.CodeAnalysis.Host.Mef.MefHostServices.Create(IEnumerable`1 assemblies) in /_/src/Workspaces/Core/Portable/Workspace/Host/Mef/MefHostServices.cs:line 55
   at Microsoft.CodeAnalysis.Host.Mef.MefHostServices.get_DefaultHost() in /_/src/Workspaces/Core/Portable/Workspace/Host/Mef/MefHostServices.cs:line 88
   at Microsoft.CodeAnalysis.AdhocWorkspace..ctor() in /_/src/Workspaces/Core/Portable/Workspace/AdhocWorkspace.cs:line 27
   at Microsoft.EntityFrameworkCore.Design.Internal.CSharpHelper..ctor(ITypeMappingSource typeMappingSource)
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Void** arguments, Signature sig, Boolean isConstructor)
   at System.Reflection.MethodBaseInvoker.InvokeDirectByRefWithFewArgs(Object obj, Span`1 copyOfArgs, BindingFlags invokeAttr)
   at System.Reflection.MethodBaseInvoker.InvokeWithOneArg(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteVisitor`2.VisitCallSiteMain(ServiceCallSite callSite, TArgument argument)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitRootCache(ServiceCallSite callSite, RuntimeResolverContext context)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteVisitor`2.VisitCallSite(ServiceCallSite callSite, TArgument argument)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitConstructor(ConstructorCallSite constructorCallSite, RuntimeResolverContext context)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteVisitor`2.VisitCallSiteMain(ServiceCallSite callSite, TArgument argument)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitRootCache(ServiceCallSite callSite, RuntimeResolverContext context)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteVisitor`2.VisitCallSite(ServiceCallSite callSite, TArgument argument)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitConstructor(ConstructorCallSite constructorCallSite, RuntimeResolverContext context)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteVisitor`2.VisitCallSiteMain(ServiceCallSite callSite, TArgument argument)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitRootCache(ServiceCallSite callSite, RuntimeResolverContext context)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteVisitor`2.VisitCallSite(ServiceCallSite callSite, TArgument argument)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitIEnumerable(IEnumerableCallSite enumerableCallSite, RuntimeResolverContext context)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteVisitor`2.VisitCallSiteMain(ServiceCallSite callSite, TArgument argument)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitRootCache(ServiceCallSite callSite, RuntimeResolverContext context)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteVisitor`2.VisitCallSite(ServiceCallSite callSite, TArgument argument)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitConstructor(ConstructorCallSite constructorCallSite, RuntimeResolverContext context)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteVisitor`2.VisitCallSiteMain(ServiceCallSite callSite, TArgument argument)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitRootCache(ServiceCallSite callSite, RuntimeResolverContext context)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteVisitor`2.VisitCallSite(ServiceCallSite callSite, TArgument argument)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitConstructor(ConstructorCallSite constructorCallSite, RuntimeResolverContext context)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteVisitor`2.VisitCallSiteMain(ServiceCallSite callSite, TArgument argument)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitCache(ServiceCallSite callSite, RuntimeResolverContext context, ServiceProviderEngineScope serviceProviderEngine, RuntimeResolverLock lockType)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitScopeCache(ServiceCallSite callSite, RuntimeResolverContext context)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteVisitor`2.VisitCallSite(ServiceCallSite callSite, TArgument argument)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitConstructor(ConstructorCallSite constructorCallSite, RuntimeResolverContext context)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteVisitor`2.VisitCallSiteMain(ServiceCallSite callSite, TArgument argument)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitCache(ServiceCallSite callSite, RuntimeResolverContext context, ServiceProviderEngineScope serviceProviderEngine, RuntimeResolverLock lockType)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitScopeCache(ServiceCallSite callSite, RuntimeResolverContext context)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteVisitor`2.VisitCallSite(ServiceCallSite callSite, TArgument argument)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.Resolve(ServiceCallSite callSite, ServiceProviderEngineScope scope)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.DynamicServiceProviderEngine.<>c__DisplayClass2_0.<RealizeService>b__0(ServiceProviderEngineScope scope)
   at Microsoft.Extensions.DependencyInjection.ServiceProvider.GetService(ServiceIdentifier serviceIdentifier, ServiceProviderEngineScope serviceProviderEngineScope)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.ServiceProviderEngineScope.GetService(Type serviceType)
   at Microsoft.Extensions.DependencyInjection.ServiceProviderServiceExtensions.GetRequiredService(IServiceProvider provider, Type serviceType)
   at Microsoft.Extensions.DependencyInjection.ServiceProviderServiceExtensions.GetRequiredService[T](IServiceProvider provider)
   at Microsoft.EntityFrameworkCore.Design.Internal.MigrationsOperations.AddMigration(String name, String outputDir, String contextType, String namespace)
   at Microsoft.EntityFrameworkCore.Design.OperationExecutor.AddMigrationImpl(String name, String outputDir, String contextType, String namespace)
   at Microsoft.EntityFrameworkCore.Design.OperationExecutor.AddMigration.<>c__DisplayClass0_0.<.ctor>b__0()
   at Microsoft.EntityFrameworkCore.Design.OperationExecutor.OperationBase.<>c__DisplayClass3_0`1.<Execute>b__0()
   at Microsoft.EntityFrameworkCore.Design.OperationExecutor.OperationBase.Execute(Action action)
System.TypeLoadException: Method 'get_IsParamsArray' in type 'Microsoft.CodeAnalysis.CodeGeneration.CodeGenerationParameterSymbol' from assembly 'Microsoft.CodeAnalysis.Workspaces, Version=4.5.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35' does not have an implementation.
Unable to load one or more of the requested types.
Method 'get_IsParamsArray' in type 'Microsoft.CodeAnalysis.CodeGeneration.CodeGenerationParameterSymbol' from assembly 'Microsoft.CodeAnalysis.Workspaces, Version=4.5.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35' does not have an implementation.
```

**Comment by JeremyCaron** [comment]
@canselcik See my comment above, I just edited it with a description of how to rollback the C# extension to 2.22.5 which worked for me.

**Comment by canselcik** [comment]
> @canselcik See my comment above, I just edited it with a description of how to rollback the C# extension to 2.22.5 which worked for me.

That worked, I appreciate it. The UI definitely doesn't help the situation.

**Comment by CyrusNajmabadi** [comment]
@sharwell is this complete?  If not, waht is left to be done?

**Comment by CyrusNajmabadi** [comment]
@sharwell is this complete? If not, waht is left to be done?

**Comment by sharwell** [comment]
@CyrusNajmabadi Yes this was fixed in 17.12

**Comment by CyrusNajmabadi** [comment]
@sharwell thanks :)
